test/integration/integration.h

```diff
@@ -220,12 +220,14 @@ class BaseIntegrationTest : Logger::Loggable<Logger::Id::testing> {
   // sending/receiving to/from the (imaginary) xDS server. You should almost always use
   // compareDiscoveryRequest() and sendDiscoveryResponse(), but the SotW/delta-specific versions are
   // available if you're writing a SotW/delta-specific test.
+  // TODO(fredlas) expect_node was defaulting false here; the delta+SotW unification work restores
+  // it.
```

Comment:
@fredlas We changed the xDS protocol _not_ to send node with every request on a stream, as the node metadata is sometimes larger than the response. I think you might want to revert this change and file a TODO to implement node skipping for delta gRPC.

Reply:
See the text of the TODO: I made sure to accommodate it in the unified version.

This PR was ready to merge a month+ before #7876 was opened, so it would make more sense for that one to be reverted, and reapplied after this one is merged. The ideal, of course, would have been becoming aware of that work much earlier, so that I could recommend it be delayed until after these changes. However, I figured trying to get an already-merged PR reverted would just make things even more slow, and the duplicated effort (since #7876 was written on top of soon-to-be-deleted code) was unfortunately already locked in. So, I reimplemented the intention of #7876 in https://github.com/envoyproxy/envoy/compare/master...fredlas:unified_ADS , which rewrites/refactors gRPC xDS to let this PR's delta additions exist cleanly alongside the existing state-of-the-world logic. Since this PR represents a functional but messy initial version, it is not worth incorporating in here.

Reply:
As long as we respect the flag for non-incremental xDS, I see no problem, since we can always add this flag to incremental version. I can't immediately tell if that's the case. We are already using this optimization in istio, so I am trying to check if there are regressions in this area.